source/beamline/SGM/monochromator/SGMEnergyTrajectory.cpp

```diff
@@ -0,0 +1,199 @@
+#include "SGMEnergyTrajectory.h"
+
+SGMEnergyTrajectory::SGMEnergyTrajectory(double startEnergy,
+                                         double endEnergy,
+                                         double time,
+                                         SGMGratingSupport::GratingTranslation gratingTranslation)
+{
+
+    startMonoInfo_ = new SGMMonochromatorInfo(startEnergy, gratingTranslation);
+    startEnergy_ = startEnergy;
+    endMonoInfo_ = new SGMMonochromatorInfo(endEnergy, gratingTranslation);
+    endEnergy_ = endEnergy;
+
+    // ensure both undulator harmonics are the same.
+    if(startMonoInfo_->undulatorHarmonic() != endMonoInfo_->undulatorHarmonic()) {
+
+        // If there is a conflict, just use the first harmonic.
```

Comment:
I'm going to speak with Tom regarding this. It's a fairly arbitrary decision, but there is probably some more complex logic which could be used to determine which harmonic to use in cases of conflict.